tools/clang_tools/README.md

```diff
@@ -0,0 +1,60 @@
+# Envoy Clang libtool developer tools
+
+## Overview
+
+A number of tools live in this directory that are intended for use by Envoy
+developers (and potentially CI). These are host tools and should not be linked
+into the Envoy target. They are based around Clang's
+[libtooling](https://clang.llvm.org/docs/LibTooling.html) libraries, a C++
+framework for writing Clang tools in the style of `clang-format` and
+`clang-check`.
+
+## Building and running
+
+To build tools in this tree, a Clang binary install must be available. If you
+are building Envoy with `clang`, this should already be true of your system. You
+can find prebuilt binary releases of Clang at https://releases.llvm.org. You
+will need the Clang version used by Envoy in CI (currently clang-9.0).
+
+To build a tool, set the following environment variables:
+
+```console
+export CLANG_TOOLS_LIB=<path to clang installation>/lib
```

Comment:
Can these be automatically detected if `llvm-config-9` command is available? `llvm-config-9 --includedir` and `llvm-config-9 --libdir` prints what you want here.

Reply:
Good idea, I'll look into this.

Reply:
Switching to LLVM_CONFIG as an input variable helps this.